README.md

```diff
@@ -1,315 +1,331 @@
+
+After git is installed, you will want to configure your git user to sign
+your commits. This can be done by following all the steps to [verify commit signatures](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification). You will also
+need to [add your GPG public key to your GitHub profile](https://github.com/settings/keys),
+and make sure that the email address set in your GitHub account matches
+the one you specified when generating your keys. You can find more information
+about this in [GitHub's documentation on commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification). To avoid having to
+type in your password all the time, you may also want to [set up SSH key
+access to your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh).
+
+After accepting the invitation on GitHub Classroom, you can clone the
+assignment repository by clicking the green "Code" button, copying the
+repository URL under "Clone", and then running:
+
+```
+git clone <your_repository_url>
+```
+
+Note that if you have set up an SSH key, you will want to make sure you
+copy the SSH URL (which looks like `git@github.com:NYUAppSec/...`).
+
+The next step is to set up GitHub Actions to automatically build and test
+your code when you push a commit. You can find a [tutorial on GitHub Actions here](https://docs.github.com/en/actions/learn-github-actions/introduction-to-github-actions).
+
+For now, you should set up GitHub Actions to just run
+
+```
+echo "Hello world"
+```
+
+Whenever a new commit is pushed to the repository, GitHub Actions will run.
+To do this, you'll create a file named `.github/workflows/hello.yml`.
+Check that the Action is running correctly in the GitHub interface.
+
+### Submission
+If you’d like to submit this part, push the `hw1p1handin` tag with the following:
+
+    git tag -a -m "Completed hw1 part1." hw1p1handin
+    git push origin main
+    git push origin hw1p1handin
+
+## Part 2: Auditing and Test Cases
+
+Read through the `giftcardreader.c` program (and its accompanying
+header, `giftcard.h`) to get a feel for it. You should also try building
+and running it with the included `examplefile.gft` file, to see what its
+normal output is. You may find it helpful to use a debugger like `gdb`
+or `lldb` to step through the program as it executes.
+
+There is also a `Makefile` included in the repository. You can use this
+to build the program by typing `make`. You will have three executables
+
+* giftcardreader.original - This is the giftcard reader without any modifications
+* giftcardreader.asan -  Thie executable has a compiler flag `-fsanitize=address` that tells the compiler to use the AddressSanitizer, a memory error detector.

```

Comment:
* giftcardreader.asan - This executable has a compiler flag `-fsanitize=address` that tells the compiler to use the AddressSanitizer, a memory error detector.